lib/azure/armrest/storage_account_service.rb

```diff
@@ -276,26 +276,31 @@ def get_private_images(storage_accounts)
             key = get_account_key(storage_account)
           rescue Azure::Armrest::ApiException
             next # Most likely due to incomplete or failed provisioning.
+          else
+            storage_account.access_key = key
           end
 
-          storage_account.all_blobs(key, configuration.max_threads).each do |blob|
-            next unless File.extname(blob.name).casecmp('.vhd') == 0
-            next unless blob.properties.lease_state.casecmp('available') == 0
-
-            # In rare cases the endpoint will be unreachable. Warn and move on.
-            begin
-              blob_properties = storage_account.blob_properties(blob.container, blob.name, key)
-            rescue Errno::ECONNREFUSED, Azure::Armrest::TimeoutException => err
-              msg = "Unable to collect blob properties for #{blob.name}/#{blob.container}: #{err}"
-              log('warn', msg)
-              next
-            end
-
-            next unless blob_properties.respond_to?(:x_ms_meta_microsoftazurecompute_osstate)
-            next unless blob_properties.x_ms_meta_microsoftazurecompute_osstate.casecmp('generalized') == 0
-
-            mutex.synchronize do
-              results << blob_to_private_image_object(storage_account, blob, blob_properties)
+          storage_account.containers.each do |container|
+            next if container.name =~ /diagnostic/i
```

Comment:
@djberg96 Who sets the name of the diagnostic container? Will it always contain the string "diagnostic"?

Reply:
@bronaghs Azure does. I guess the name is theoretically configurable after the fact, but by default it will contain the string "diagnostics" if, for example, boot diagnostics are enabled in the Azure portal.

Reply:
OK thanks